integrade/config.py

```diff
@@ -31,14 +31,46 @@ def get_config():
         _CONFIG['api_version'] = os.environ.get(
             'CLOUDIGRADE_API_VERSION', 'v1')
         _CONFIG['base_url'] = os.environ.get('CLOUDIGRADE_BASE_URL', '')
-        # expect CLOUDIGRADE_CUSTOMER_ROLE_ARNS to be a whitespace delimitted
-        # list of valid ARNs, each tied to a different AWS account
-        _CONFIG['valid_roles'] = os.environ.get(
-            'CLOUDIGRADE_CUSTOMER_ROLE_ARNS', '')
-        if len(_CONFIG['valid_roles']) > 0:
-            _CONFIG['valid_roles'] = _CONFIG['valid_roles'].split()
-        else:
-            _CONFIG['valid_roles'] = []
+        # pull all customer roles out of environ
+
+        def is_role(string): return string.startswith('CLOUDIGRADE_ROLE_')
+
+        def profile_name(string): return string.replace(
+            'CLOUDIGRADE_ROLE_', '')
+
+        profiles = [{'arn': os.environ.get(role),
+                     'name': profile_name(role)}
+                    for role in filter(is_role, os.environ.keys())
+                    ]
+        _CONFIG['aws_profiles'] = profiles
+
+        try:
+            aws_image_config = get_aws_image_config()
+        except exceptions.ConfigFileNotFoundError:
+            aws_image_config = {}
+
+        for profile in _CONFIG['aws_profiles']:
+            profile_name = profile['name'].upper()
+            acct_arn = profile['arn']
+            acct_num = [
+                num for num in filter(
+                    str.isdigit,
+                    acct_arn.split(':'))][0]
+            profile['account_number'] = acct_num
+            profile['cloudtrail_name'] = f'cloudigrade-{acct_num}'
+            profile['access_key_id'] = os.environ.get(
+                f'AWS_ACCESS_KEY_ID_{profile_name}')
+            profile['access_key'] = os.environ.get(
+                f'AWS_SECRET_ACCESS_KEY_{profile_name}')
+            profile['images'] = aws_image_config.get('profiles', {}).get(
+                profile_name, {}).get('images', [])
+
+            if not profile['access_key_id']:
```

Comment:
DO you think it would be better to collect all missing config and raise the exception stating all missing config at once? That may be useful when running the automation or running the first time to have a single fail with many missing things that need to run many times to discover new items.

Reply:
let me look at this some more, thanks for the suggestion